Assets/MixedRealityToolkit.SDK/Features/UX/Scripts/BoundingBox/BoundingBox.cs

```diff
@@ -1536,158 +1566,121 @@ private void SetBoundingBoxCollider()
+                    if (AddColliderBoundsToTarget(colliderByTransform)) { continue; }
+                    if (boundsCalculationMethod == BoundsCalculationMethod.ColliderOnly) { continue; }
                 }
 
-                GetCornerPositionsFromBounds(bounds, ref boundsCorners);
-                for (int c = 0; c < boundsCorners.Length; ++c)
+                // Encapsulate the renderer bounds if criteria match
+
+                if (boundsCalculationMethod != BoundsCalculationMethod.ColliderOnly)
                 {
-                    GameObject cube = GameObject.CreatePrimitive(PrimitiveType.Cube);
-                    cube.name = c.ToString();
-                    cube.transform.localScale = new Vector3(0.02f, 0.02f, 0.02f);
-                    cube.transform.position = boundsCorners[c];
+                    if (AddRendererBoundsToTarget(rendererBoundsByTransform)) { continue; }
+                    if (boundsCalculationMethod == BoundsCalculationMethod.RendererOnly) { continue; }
                 }
 
-                boundsMethod = BoundsCalculationMethod.Renderers;
-                return bounds;
+                // Do the collider for the one case that we chose RendererOverCollider and did not find a renderer
+                AddColliderBoundsToTarget(colliderByTransform);
             }
 
-            MeshFilter[] meshFilters = Target.GetComponentsInChildren<MeshFilter>();
-            if (meshFilters.Length > 0)
+            if (colliderCorners.Count == 0) { return new Bounds(); }
+
+            Transform targetTransform = Target.transform;
+
+            Bounds finalBounds = new Bounds(targetTransform.InverseTransformPoint(colliderCorners[0]), Vector3.zero);
+
+            for (int i = 1; i < colliderCorners.Count; i++)
             {
-                //Mesh.bounds is local space bounding volume
-                bounds.size = meshFilters[0].mesh.bounds.size;
-                bounds.center = meshFilters[0].mesh.bounds.center;
-                for (int i = 0; i < meshFilters.Length; ++i)
-                {
-                    bounds.Encapsulate(meshFilters[i].mesh.bounds);
-                }
-                if (bounds.size != Vector3.zero)
-                {
-                    bounds.center = Target.transform.position;
-                    boundsMethod = BoundsCalculationMethod.MeshFilters;
-                    return bounds;
-                }
+                finalBounds.Encapsulate(targetTransform.InverseTransformPoint(colliderCorners[i]));
             }
 
-            BoxCollider boxCollider = Target.AddComponent<BoxCollider>();
-            bounds = boxCollider.bounds;
-            Destroy(boxCollider);
-            boundsMethod = BoundsCalculationMethod.Collider;
-            return bounds;
+            return finalBounds;
         }
 
-        private Bounds GetSingleObjectBounds(GameObject gameObject)
+        private bool AddRendererBoundsToTarget(KeyValuePair<Transform, Bounds> rendererBoundsByTarget)
```

Comment:
In MRTK when we try to do something and then return bool if succeeded / failed, we normally use TryXYZ, could you please rename to TryAddRendererBoundsToTarget?

Reply:
not returning a bool anymore, therefore skipped